### PR TITLE
 New option for setting the 'required' attribute on the date picker input element.

### PR DESCRIPTION
**Pull request by scrappergus**
I've added a simple boolean attribute to the Datepicker that allows you to set the input element's 'required' attribute.

In order to pass the pre-commit hooks, I had to disable the NextJS-related ESLint plugins when committing, and also change the line order in a couple files.

### CODE REVIEW

**Comment by alphabetek** [comment]
@onesine any chance to land this PR? Thank you in advance.

**Comment by netgfx** [comment]
Looking for this as well

**Comment by idevstuffs** [comment]
Great addition! I'm looking for this feature as well, please approve and merge!

**Comment by onesine** [comment]
Hello @scrappergus, @alphabetek, @idevstuffs 👋.
Sorry for the late reply.
Thanks for the contribution @scrappergus.

File: .eslintrc.json [hunk 1]
```diff
@@ -10,8 +10,7 @@
         "plugin:react/recommended",
         "plugin:@typescript-eslint/recommended",
         "plugin:prettier/recommended",
-        "plugin:react-hooks/recommended",
-        "next/core-web-vitals"
```
Comment (onesine) [inline .eslintrc.json:14]: Why remove this plugin

File: .eslintrc.json [hunk 2]
```diff
@@ -27,7 +26,7 @@
             "version": "detect"
         }
     },
-    "plugins": ["react", "@typescript-eslint", "import", "prettier", "@next/eslint-plugin-next"],
```
Comment (onesine) [inline .eslintrc.json:30]: Why remove this plugin

**Comment by scrappergus** [comment]
The commit was failing pre-commit hooks, so in order to create the pull-request, I had to.

Gus Clemens
***@***.***

**Pull request merged**